loader.go

```diff
@@ -39,22 +42,56 @@ var _ Loader = (*OSFileSystemLoader)(nil)
 // NewOSFileSystemLoader returns an initialized OSFileSystemLoader.
 func NewOSFileSystemLoader(dirPath string) *OSFileSystemLoader {
 	return &OSFileSystemLoader{
-		dir: dirPath,
+		dir: filepath.FromSlash(dirPath),
 	}
 }
 
 // Open opens a file from OS file system.
 func (l *OSFileSystemLoader) Open(path string) (io.ReadCloser, error) {
-	return os.Open(filepath.Join(l.dir, path))
+	return os.Open(filepath.Join(l.dir, filepath.FromSlash(path)))
 }
 
 // Exists checks if the template name exists by walking the list of template paths
 // returns true if the template file was found
 func (l *OSFileSystemLoader) Exists(path string) (string, bool) {
-	path = filepath.Join(l.dir, path)
+	path = filepath.Join(l.dir, filepath.FromSlash(path))
 	stat, err := os.Stat(path)
 	if err == nil && !stat.IsDir() {
 		return path, true
 	}
 	return "", false
 }
+
+type InMemLoader struct {
+	files map[string][]byte
+}
+
+// compile time check that we implement Loader
+var _ Loader = (*InMemLoader)(nil)
+
+func NewInMemLoader() *InMemLoader {
+	return &InMemLoader{
+		files: map[string][]byte{},
```

Comment:
Accessing `l.files` without mutex allows for race conditions. There a few areas within Jet that could use a second look (see also: https://github.com/CloudyKit/jet/issues/181)

Reply:
will be addressed in fix for #181